internal/konnect/client.go

```diff
@@ -0,0 +1,179 @@
+package konnect
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/adminapi"
+	tlsutil "github.com/kong/kubernetes-ingress-controller/v2/internal/util/tls"
+)
+
+// Client is used for sending requests to Konnect APIs which are not included
+// in Kong Admin APIs, like node registration APIs or runtime group operation APIs.
+// TODO(naming): give a better type name to this client?
+type Client struct {
+	Address        string
+	RuntimeGroupID string
+	Client         *http.Client
+}
+
+// KicNodeAPIPathPattern is the path pattern for KIC node operations.
+var KicNodeAPIPathPattern = "%s/kic/api/runtime_groups/%s/v1/kic-nodes"
+
+// NewClient creates a Konnect client.
+func NewClient(cfg adminapi.KonnectConfig) (*Client, error) {
+	tlsConfig := tls.Config{ //nolint:gosec
```

Comment:
Can't we satisfy gosec's complaints and not add the `nolint` directive here?

Reply:
set the `tls.MinVersion`.